sycamore/reader.py

```diff
@@ -33,6 +33,25 @@ def binary(
         )
         return DocSet(self._context, scan)
 
+    def manifest(
+        self,
+        metadata_provider: FileMetadataProvider,
+        binary_format: str,
+        parallelism: Optional[int] = None,
+        filesystem: Optional[FileSystem] = None,
+        **resource_args
+    ) -> DocSet:
+        paths = metadata_provider.get_paths()
+        scan = BinaryScan(
+            paths,
+            binary_format=binary_format,
+            parallelism=parallelism,
+            filesystem=filesystem,
+            metadata_provider=metadata_provider,
+            **resource_args
+        )
+        return DocSet(self._context, scan)
+
```

Comment:
would this work for html/non-binary?

Reply:
It will work with html as we are still using Binary scan for HTMLs. We can add support for non-binaries as needed.